### PR TITLE
Return error if no inputs / outputs are provided in model config

### PR DESCRIPTION
**Pull request by CoderHam**
@deadeyegoodwin we now report an error if there are no inputs or outputs. However the ValidateInputs/ValidateOutputs functions do not check for the presence of inputs/outputs.

 I believe we should enforce this checks for IO validation.

### CODE REVIEW

File: src/libtorch.cc [hunk 1]
```diff
@@ -225,6 +225,29 @@ ModelState::AutoCompleteConfig()
...
+      output_cnt = outputs.ArraySize();
+    }
+  }
+
+  if ((input_cnt == 0) || (output_cnt == 0)) {
+    return TRITONSERVER_ErrorNew(
+        TRITONSERVER_ERROR_INTERNAL,
+        "model configuration must contain inputs and outputs since "
+        "auto-complete is not supported for pytorch backend");
+  }
```
Comment (GuanLuo) [inline src/libtorch.cc:249]: The above should be remove as the validate functions are already responsible for checking if there are I/Os. Nothing should be done in autofill as PyTorch model doesn't provide model information.

File: src/libtorch.cc [hunk 2]
```diff
@@ -501,7 +501,7 @@ ModelInstanceState::ModelInstanceState(
     }
   }
 
-  THROW_IF_BACKEND_INSTANCE_ERROR(ValidateInputs());
+  THROW_IF_BACKEND_INSTANCE_ERROR(ValidateInputs(expected_input_cnt));
```
Comment (GuanLuo) [inline src/libtorch.cc:504]: That won't be correct as `expected_input_cnt` looks at the model config already. So no input results in  `expected_input_cnt == 0` and validate function won't report error correctly

Comment (CoderHam) [reply src/libtorch.cc:504]: We have an additional check for when no inputs are specified: https://github.com/triton-inference-server/pytorch_backend/blob/77adac248889f21035eb57efcae5eae8d65ef5c0/src/libtorch.cc#L604-L608

Using expected_input_cnt here is symmetric with how the [onnxruntime backend](https://github.com/triton-inference-server/onnxruntime_backend/blob/main/src/onnxruntime.cc#L1134-L1141) does it.

Comment (CoderHam) [reply src/libtorch.cc]: Removed this check since we cannot use it for pytorch backend

Comment (GuanLuo) [reply src/libtorch.cc]: In ORT it is comparing the model config to the model metadata, which is not the same in PyTorch

**Comment by GuanLuo** [comment]
Can you add this edge case to L0_model_config?

**Comment by CoderHam** [comment]
> Can you add this edge case to L0_model_config?

https://github.com/triton-inference-server/server/pull/3600

**Review by GuanLuo — APPROVED** [review]

**Pull request merged**